### PR TITLE
Update resource useing generator script 2

### PR DESCRIPTION
**Pull request by myakove**

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: scripts/resource/class_generator.py [hunk 1]
```diff
@@ -392,136 +392,84 @@ def parse_explain(
...
-        # If line is empty section is done
-        if not line.strip():
-            if section_data:
-                sections.append(section_data)
-                section_data = ""
+    raw_resource_dict: Dict[str, str] = {}
 
-            continue
+    # Get all sections from output, section is [A-Z]: for example `KIND:`
+    sections = re.findall(r"([A-Z]+:).*", output)
```
Comment (rnetser) [inline scripts/resource/class_generator.py:403]: will moving ":" outside the group remove the need to strip it?
`re.findall(r"([A-Z]+):.*", output)`

File: scripts/resource/class_generator.py [hunk 2]
```diff
@@ -392,136 +392,84 @@ def parse_explain(
...
+    # Get all spec fields till spec indent is done, section indent is 2 empty spaces
+    # ```
+    #  spec  <ServiceSpec>
+    #    allocateLoadBalancerNodePorts       <boolean>
+    #    type        <string>
+    #  status        <ServiceStatus>
+    # ```
+    if _spec_fields := re.findall(rf"  {SPEC_STR.lower()}.*(?=\n  [a-z])", raw_resource_dict[FIELDS_STR], re.DOTALL):
+        for field in [_field for _field in _spec_fields[0].splitlines() if _field]:
+            # If line is indented 4 spaces we know that this is a field under space
```
Comment (rnetser) [inline scripts/resource/class_generator.py:440]:
```suggestion
            # If line is indented 4 spaces we know that this is a field under spec
```
?

File: scripts/resource/class_generator.py [hunk 3]
```diff
@@ -392,136 +392,84 @@ def parse_explain(
...
-                                output_debug_file_path=output_debug_file_path,
-                                add_tests=add_tests,
-                            )
-                        )
-                        continue
-
-            else:
-                break
-
+    # TODO: Do we have resources without `spec` or user configurable fields?
```
Comment (rnetser) [inline scripts/resource/class_generator.py:472]: do you mean like `ClusterOperator` which looks like this:
```
metadata....
  spec  <Object> -required-
  status        <Object>
...
```
?

Comment (myakove) [reply scripts/resource/class_generator.py:472]: I mean resource that do not have to_dict, which the user cannot create, only get.

Comment (rnetser) [reply scripts/resource/class_generator.py:472]: so ClusterOperator is an example

Comment (rnetser) [reply scripts/resource/class_generator.py:472]: do you want to address it in this PR? add the name so we can address it later?

**Review by rnetser — DISMISSED** [review]

File: scripts/resource/class_generator.py (hunk 3 above)
Comment (myakove) [reply scripts/resource/class_generator.py:472]: I will do it in this PR

File: class_generator/manifests/class_generator_template.j2 [hunk 4]
```diff
@@ -25,20 +25,25 @@ class {{ KIND }}({{ BASE_CLASS }}):
 
     def __init__(
         self,
+        {% if FIELDS or SPEC %}
```
Comment (rnetser) [inline class_generator/manifests/class_generator_template.j2:28]:
```suggestion
        {% if all_types_for_class_args %}
```
?

**Review by rnetser — APPROVED** [review]

**Pull request merged**